repository/src/main/java/org/apache/atlas/repository/store/graph/v2/tags/TagDAOCassandraImpl.java

```diff
@@ -274,18 +278,38 @@ public void putPropagatedTags(String sourceAssetId, String tagTypeName, Set<Stri
     }
 
     @Override
-    public List<AtlasClassification> getAllDirectTagsForVertex(String vertexId) throws AtlasBaseException {
-        AtlasPerfMetrics.MetricRecorder recorder = RequestContext.get().startMetricRecord("getAllDirectTagsForVertex");
-        List<AtlasClassification> tags = new ArrayList<>();
+    public List<AtlasClassification> getAllDirectClassificationsForVertex(String vertexId) throws AtlasBaseException {
```

Comment:
I think we should simplify this with dynamic columns population, we have multiple methods for each combination of columns selection, also needs multiple statements.

Check on AI tools if we can simplify this with less number of prepared statements & less methods in TagDAOCassandraImpl class